src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,92 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int INITIAL_CAPACITY = 16;
+    private Node<K, V>[] table;
+    private int size;
+
+    // Внутрішній клас для представлення пари ключ-значення
+    private static class Node<K, V> {
+        private K key;
+        private V value;
+        private Node<K, V> next;
+
+        Node(K key, V value, Node<K, V> next) {
+            this.key = key;
+            this.value = value;
+            this.next = next;
+        }
+    }
+
+    public MyHashMap() {
+        table = new Node[INITIAL_CAPACITY];
+        size = 0;
+    }
 
     @Override
     public void put(K key, V value) {
+        if (size >= table.length * 0.75) {
+            resize();
+        }
+
+        int index = (key == null) ? 0 : Math.abs(key.hashCode() % table.length);
```

Comment:
The calculation of the index using `Math.abs(key.hashCode() % table.length)` can lead to negative indices if `key.hashCode()` returns `Integer.MIN_VALUE`. Consider using `(key.hashCode() & 0x7fffffff) % table.length` to ensure a non-negative index.